substratevm/src/com.oracle.svm.truffle/src/com/oracle/svm/truffle/TruffleBaseFeature.java

```diff
@@ -625,6 +628,62 @@ private static void registerReflectionAccessesForRuntimeValidation(Class<?> stor
     }
 }
 
+@TargetClass(className = "com.oracle.truffle.api.staticobject.StaticProperty", onlyWith = TruffleBaseFeature.IsEnabled.class)
+final class Target_com_oracle_truffle_api_staticobject_StaticProperty {
+    public static final class OffsetTransformer implements RecomputeFieldValue.CustomFieldValueTransformer {
+
+        private static final int svmArrayBaseOffset;
+        private static final int svmArrayIndexScale;
+
+        // We have to use reflection to access private members instead of aliasing them
+        // in the substitution class since substitutions are present only in runtime
+        private static final Method staticPropertyKindGetInternalKind;
+        private static final byte objectStaticPropertyKindByte;
+
+        static {
+            try {
+                // Checkstyle: stop
+                Class<?> staticPropertyKindClass = Class.forName("com.oracle.truffle.api.staticobject.StaticPropertyKind");
+                staticPropertyKindGetInternalKind = ReflectionUtil.lookupMethod(StaticProperty.class, "getInternalKind");
+                Object objectStaticPropertyKind = ReflectionUtil.readField(staticPropertyKindClass, "Object", null);
+                Method staticPropertyKindToByte = ReflectionUtil.lookupMethod(staticPropertyKindClass, "toByte");
+                objectStaticPropertyKindByte = (byte) staticPropertyKindToByte.invoke(objectStaticPropertyKind);
+                // Checkstyle: resume
+            } catch (ReflectiveOperationException e) {
+                throw VMError.shouldNotReachHere(e);
+            }
+
+            // Find SVM array base offset and array index scale
+            svmArrayBaseOffset = ConfigurationValues.getObjectLayout().getArrayBaseOffset(JavaKind.Object);
+            svmArrayIndexScale = ConfigurationValues.getObjectLayout().getArrayIndexScale(JavaKind.Object);
+        }
+
+        @Override
+        public Object transform(MetaAccessProvider metaAccess, ResolvedJavaField original, ResolvedJavaField annotated,
+                        Object receiver, Object originalValue) {
+            int offset = (int) originalValue;
+            StaticProperty receiverStaticProperty = (StaticProperty) receiver;
+            try {
+                byte internalKindByte = (byte) staticPropertyKindGetInternalKind.invoke(receiverStaticProperty);
+
+                // Re-computation is performed only if the kind is Object:
```

Comment:
We use `/**/` style comments.

Reply:
Changed. I also felt free to update the rest of the comments in this file that did not follow this convention.